### ISSUE TITLE
Templates break root lookup

### ISSUE THREAD

**Issue by thegedge** [sunng87/handlebars-rust#264]
Verified in http://tryhandlebarsjs.com/:

```rust
#[test]
fn test_root_with_blocks() {
    let hbs = Handlebars::new();

    let data = json!({
        "a": [
            {"b": 1},
            {"b": 2},
        ],
        "b": 3,
    });

    let template = "{{#*inline \"test\"}}{{b}}:{{@root.b}};{{/inline}}{{#each a}}{{> test}}{{/each}}";
    assert_eq!(
        hbs.render_template(template, &data).unwrap(),
        "1:3;2:3;"
    );
}
```

```
failures:

---- test_root_with_blocks stdout ----
thread 'test_root_with_blocks' panicked at 'assertion failed: `(left == right)`
  left: `"1:1;2:2;"`,
 right: `"1:3;2:3;"`', tests/block_context.rs:17:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Comment by thegedge** [comment]
Actually, this works fine in `master`, but is broken only in https://github.com/sunng87/handlebars-rust/pull/263

**Comment by sunng87** [comment]
For issues with PRs in review, you can leave comment under the PR 😀. This defeat is caused by changes in partial context. Together with #263 I will look into block param handling to correct all these behaviours.

**Comment by thegedge** [comment]
> For issues with PRs in review, you can leave comment under the PR 😀

Haha, next time I will. I posted this issue before realizing I was working from your PR branch.

Thanks so much for taking on this work!

**Comment by sunng87** [comment]
Fixed in #263
